### PR TITLE
Payment filter model

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
Listening to paymentFilterStream
Applying payment filter
Change paymentsStream to return PaymentsModel

### CODE REVIEW

File: lib/bloc/account/account_bloc.dart [hunk 1]
```diff
@@ -45,15 +45,12 @@ class AccountBloc {
...
-  Stream<List<PaymentInfo>> get paymentsStream => _paymentsController.stream;
+  final _paymentsController = new BehaviorSubject<PaymentsModel>();
+  Stream<PaymentsModel> get paymentsStream => _paymentsController.stream;
 
-  Stream<List<PaymentInfo>> get receivedPayments {
-    return paymentsStream.map( (payments) => payments.where( (p) => [PaymentType.DEPOSIT, PaymentType.RECEIVED].contains(p.type)));
-  }
-  Stream<List<PaymentInfo>> get sentPayments {
-    return paymentsStream.map( (payments) => payments.where( (p) => [PaymentType.WITHDRAWAL, PaymentType.SENT].contains(p.type)));
-  }
```
Comment (erdemyerebasmaz) [inline lib/bloc/account/account_bloc.dart]: Do we need or use these streams?

File: lib/bloc/account/account_bloc.dart [hunk 2]
```diff
@@ -200,19 +199,67 @@ class AccountBloc {
...
+         }
+         _paymentsController.add(PaymentsModel(_filterPayments(payments), _paymentFilterController.value, _firstDate));
       })
-      .catchError(_paymentsController.addError); 
-    }  
-  
+      .catchError(_paymentsController.addError);
+    }
+
+  _filterMockDataPayments(List<PaymentInfo> paymentsList) {
```
Comment (roeierez) [inline lib/bloc/account/account_bloc.dart]: We don't need this function. use the same function _filterPayments for the real data and mock data.
You will need to change its prototype to get List<PaymentInfo> but it worth the value of removing this function and unifying the flows.

File: lib/routes/user/home/account_page.dart [hunk 3]
```diff
@@ -49,6 +49,8 @@ class _AccountPageState extends State<_AccountPage> {
   StreamSubscription<AccountModel> _statusSubscription;
   String _filter;  
   String _paymentRequestInProgress;
+  PaymentFilterModel _paymentFilterModel;
```
Comment (roeierez) [inline lib/routes/user/home/account_page.dart]: don't save this value in the class, use it locally in the stream in the build function.

File: lib/routes/user/home/account_page.dart [hunk 4]
```diff
@@ -49,6 +49,8 @@ class _AccountPageState extends State<_AccountPage> {
...
+  DateTime _firstDate;
```
Comment (roeierez) [inline lib/routes/user/home/account_page.dart]: Also here, don't save.

**Pull request merged**